dataset/gedi_rasterization.py

```diff
@@ -0,0 +1,103 @@
+
+import ee
+import json
+import os
+import datetime
+import time
+from ee.batch import Export
+import ast
+
+def make_timestampFromYMD(year, month, day):
+    tdat = datetime.datetime(year, month, day)
+    return time.mktime(tdat.timetuple()) * 1000
+
+def rasterize_GEDI_by_utm_zone(orbits, grid_id, month):
+    props = ["orbit", "track", "beam", "channel", "degrade", "dtime", "quality",
+             "rx_algrunflag", "rx_quality", "sensitivity", "toploc", "zcross",
+             "rh10", "rh20", "rh30", "rh40", "rh50", "rh60", "rh70", "rh80", "rh90", "rh98"]
+
+    def updateOrbit(f):
+        return (f.set('orbit', fshots.get('orbit'))
+                 .set('track', fshots.get('track'))
+                )
+
+    shots = []
+    for orbit in orbits:
+        fshots = ee.FeatureCollection(orbit).filterMetadata('quality', 'equals', 1)
+        augmented = fshots.map(updateOrbit)
+        shots.append(augmented)
+
+    shots = ee.FeatureCollection(shots).flatten()
+
+    grid = (ee.FeatureCollection("users/yang/GEETables/GEDI/GEDI_UTM_GRIDS_LandOnly")
+              .filterMetadata('grid_id', 'equals', grid_id))
+
+    zone_id = ee.Feature(grid.first()).get('grid_name').getInfo()
+    crs = ee.Feature(grid.first()).get('crs').getInfo()
+
+    iprop = {
+        "month": month,
+        "year": 2019,
+        "grid_id": grid_id,
+        "version": 1,
+        "system:time_start": make_timestampFromYMD(2019, month, 15)
```

Comment:
would system:time_start of the first day of the month and system:time_end of the first day of the next month be more appropriate?

Reply:
Yes.